contracts/lib/Poseidon.sol

```diff
@@ -1,21 +1,28 @@
-pragma solidity ^0.6.0;
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
 
-contract PoseidonUnit {
-  function poseidon(uint256[] memory) public pure returns(uint256) {}
+contract PoseidonUnit2 {
+  function poseidon(uint256[2] memory) public view returns (uint256) {}
+}
+
+contract PoseidonUnit3 {
+  function poseidon(uint256[3] memory) public view returns (uint256) {}
 }
 
 contract Poseidon {
-  PoseidonUnit poseidonUnit;
+  PoseidonUnit2 _poseidonUnit2;
+    PoseidonUnit3 _poseidonUnit3;
 
-  constructor( address _poseidonContractAddr) public {
-    poseidonUnit = PoseidonUnit(_poseidonContractAddr);
+  constructor( address _poseidon2ContractAddr, address _poseidon3ContractAddr ) {
+    _poseidonUnit2 = PoseidonUnit2(_poseidon2ContractAddr);
+    _poseidonUnit3 = PoseidonUnit3(_poseidon3ContractAddr);
   }
 
-  function GetScalarField () internal pure returns (uint256){
-    return 0x30644e72e131a029b85045b68181585d2833e84879b9709143e1f593f0000001;
```

Comment:
what was the purpose of GetScalarField ?

Reply:
I have no idea. Some client code may need it for something in the past but definitely it is not used in our current system. Any reasons to bring it back?

Reply:
@OBrezhniev was it for some optimization for verifier smart contracts?